WordPress/Classes/ViewRelated/Reader/ReaderPostsViewController.m

```diff
@@ -915,7 +888,11 @@ - (NSPredicate *)predicateForFetchRequest
 
 - (NSManagedObjectContext *)managedObjectContext
 {
-    return [[ContextManager sharedInstance] mainContext];
+    if (!self.displayContext) {
+        self.displayContext = [[NSManagedObjectContext alloc] initWithConcurrencyType:NSMainQueueConcurrencyType];
```

Comment:
I think this should work just fine. You may want to create an issue to extract this into a method on ContextManager to create a child main queue context. Bottling all of the context creating into that class makes the most sense in case we start doing some weird initialization tweaks then they're all centralized.